pkg/prometheus/operator.go

```diff
@@ -1000,7 +1000,9 @@ func (c *Operator) enqueueForNamespace(store cache.Store, nsName string) {
 	}
 	ns := nsObject.(*v1.Namespace)
 
-	err = cache.ListAll(c.promInf.GetStore(), labels.Everything(), func(obj interface{}) {
+	objs, err := c.promInfs.List(labels.Everything())
```

Comment:
we should check the return error before looping over the returned values

Reply:
agreed, cherry-pick friendliness as an excuse here, but i think git will be smart enough. let me pull the err check above.

Reply:
fixed